src/hotspot/share/gc/shenandoah/shenandoahUncommitThread.hpp

```diff
@@ -66,6 +60,10 @@ class ShenandoahUncommitThread : public ConcurrentGCThread {
   // True if the control thread has allowed this thread to uncommit regions
   bool is_uncommit_allowed() const;
 
+  // Iterate and uncommit eligible regions. Return the number of regions uncommitted.
+  // This operation may be interrupted if the GC calls `forbid_uncommit`.
+  size_t do_uncommit_work(double shrink_before, size_t shrink_until) const;
```

Comment:
I'd document the semantics of the parameters too:
```
// Iterate over and uncommit eligible regions unless committed heap would fall below `shrink_until` <units: bytes?>.
// A region is eligible if it's been empty for at least `shrink_before` <units: seconds?>
// Returns the number of regions uncommitted. May be interrupted by `forbid_uncommit`.
```

Reply:
Done.